### PR TITLE
add setnx to redis

### PR DESCRIPTION
**Pull request by hdou731**

### CODE REVIEW

File: misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt [hunk 1]
```diff
@@ -71,4 +71,14 @@ class FakeRedis : Redis {
       expiryInstant = clock.instant().plusSeconds(expiryDuration.seconds)
     )
   }
+
+  override fun setnx(key: String, value: ByteString) {
+    // If the key already exists don't overwrite it
+    if (get(key) != null) return
+    // Set the key to expire at the latest possible instant
+    keyValueStore[key] = Value(
```
Comment (frojasg) [inline misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt:79]: nit: you can use `keyValueStore .putIfAbsent`

File: misk-redis/src/main/kotlin/misk/redis/RealRedis.kt [hunk 2]
```diff
@@ -60,6 +60,13 @@ class RealRedis(private val jedisPool: JedisPool) : Redis {
     }
   }
 
+  // Set a ByteArray value if it doesn't already exist
+  override fun setnx(key: String, value: ByteString) {
```
Comment (frojasg) [inline misk-redis/src/main/kotlin/misk/redis/RealRedis.kt:64]: could you add this method accepting a TTL value
```
override fun setnx(key: String, expiryDuration: Duration, value: ByteString)
```

File: misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt (hunk 1 above)
Comment (hdou731) [reply misk-redis/src/main/kotlin/misk/redis/FakeRedis.kt:79]: done

File: misk-redis/src/main/kotlin/misk/redis/RealRedis.kt (hunk 2 above)
Comment (hdou731) [reply misk-redis/src/main/kotlin/misk/redis/RealRedis.kt:64]: so the setnx function doesn't have an expiration field

Comment (hdou731) [reply misk-redis/src/main/kotlin/misk/redis/RealRedis.kt:64]: at least the one in jedis doesn't

**Review by frojasg — APPROVED** [review]

**Pull request merged**